canopeum_frontend/src/config/apiSchemaGenerator.js

```diff
@@ -0,0 +1,19 @@
+import * as fs from 'fs'
+import * as path from 'path'
+var schemaUrl = 'http://127.0.0.1:8000/api/schema/'
+var outputPath = path.join(process.cwd(), '/src/config/openapi.yaml')
+console.log('Fetching schema from:', schemaUrl)
+console.log('Saving to:', outputPath)
+fetch(schemaUrl)
+  .then(function(response) {
+    if (!response.ok) {
+      throw new Error('Failed to fetch schema')
+    }
+    return response.text()
+  })
+  .then(function(response) {
+    fs.writeFileSync(outputPath, response)
+  })
+  .catch(function(error) {
+    console.error('Failed to fetch schema:', error)
+  })
```

Comment:
You can run nswag from this script, simplifying the package.json command. Something like this:
```js
import { exec } from 'node:child_process'

...

const nswagPath = path.join(__dirname, '../../..', 'docs/canopeum.nswag')

...

fs.writeFileSync(outputPath, text)
exec(`nswag run ${nswagPath}}`)
```

Reply:
You could even delete the openapi.yaml file to not commit it to source. However, the NSwag CLI has... issues as I've just discovered trying this out:

> NSwag's initial command returns almost imediatly with a bogus "System.IO.FileNotFoundException: Could not find file" error, but a child-process is spawned that succeed and does the console printing
> So we can,t use sync commands, can't check the output, and have to manually watch for file change.

Trying to wait in the script will also block subprocesses. So our only hope would be a file-watcher to check for updates to the api.ts before deleting

Or at that point, just add `openapi.yaml` to the `.gitignore` and you're good to go.